springwolf-core/src/test/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/AsyncListenerAnnotationScannerTest.java

```diff
@@ -221,6 +259,27 @@ private void methodWithMultipleAnnotation(SimpleFoo payload) {
         }
     }
 
+    private static class ClassWithMessageAnnotation {
+
+        @AsyncListener(operation = @AsyncOperation(
+                channelName = "test-channel",
+                description = "test channel operation description"
+        ))
+        private void methodWithAnnotation(
+                @AsyncMessage(
+                        description = "Message description",
+                        messageId = "simpleFoo",
+                        name = "SimpleFooPayLoad",
+                        schemaFormat = "application/schema+json;version=draft-07",
+                        title = "Message Title"
+                ) SimpleFoo payload) {
```

Comment:
General question as I havent seen annotations with multiple fields being applied to method arguments much:
You added the annotation on the method argument, because it describes the payload and is independent of the operation (method)?

In springwolf, the `@Payload` annotation is the only one being used on parameters. While the other `AsyncListener` `KafkaAsyncOperationBinding`, etc are applied on the method.

If we apply it on the parameter, I wonder if it should replace the `@Payload` marker.

Using the AsyncMessage annotation on multiple parameters of the method (if there were), would result in undefined behaviour.

@sam0r040 What are your thoughts?

Reply:
@timonback I'm happy that you ask this question, since I asked myself the same 😄 

I decided to go with a parameter annotation because, as you mention, describes the payload. But since headers are part of the `@AsyncListener` I had second thoughts about the best place.

Reply:
@sam0r040 and I checked and we think it is best to stay closer to the AsyncApi spec and provide also a similar interface to Springwolf users.
Therefore, we would add the AsyncMessage as part of the AsyncOperation annotation
(as in the spec https://www.asyncapi.com/docs/reference/specification/v2.6.0#operationObject)
(This will probably be easier to document on our website as well)

This should avoid issues like using the annotation on multiple method parameters or mixing it with unsupported annotations like KafkaListener and RabbitListener.

And, the Payload annotation can stay as is.

Reply:
@timonback done with https://github.com/springwolf/springwolf-core/pull/231/commits/7bb542f303ec1fc027b00e3ca97868fac3f3ea18

I think that with this change all the comments are addressed